app/controllers/application_controller.rb

```diff
@@ -51,4 +52,14 @@ def complete_request(action: nil, path: nil)
   def authorize!
     raise NoMethodError
   end
+
+  def check_college
+    return if current_college.persisted? || !policy(College).edit?
```

Comment:
Because why show the flash to students / reps who won't be able to do anything about it?